source-code/app/src/main/java/org/buildmlearn/toolkit/utilities/SignerThread.java

```diff
@@ -345,7 +344,7 @@ public void attr(String ns, String name, int resourceId, int type, Object val) {
                 name = null;
             }
 
-            if (!Objects.equals(name, oldName) || val != oldVal) {
+            if ((!((name==null&&oldName==null)||(!(name==null&&oldName!=null))||(!(name!=null&&oldName==null))||(name.equals(oldName)))) || val != oldVal) {
```

Comment:
Don't make it a complicated statement. This can be reduced to a simpler one by DeMorgan's Law. 

And old statement was fine bcoz 

> Object.equals(Object a, Object b)   Returns true if the arguments are equal to each other and false otherwise. Consequently, if both arguments are null, true is returned and if exactly one argument is null, false is returned. Otherwise, equality is determined by using the equals method of the first argument.